src/java/org/apache/cassandra/service/accord/AccordService.java

```diff
@@ -592,12 +715,9 @@ public TopologyManager topology()
         }
         catch (TimeoutException e)
         {
+            // TODO (review): This is j.u.c timeout so maybe this should mark?
```

Comment:
It's double marking the metric, because `coordinateAsync` in the callback also marks the timeout. It seems like we need to be consistent about responsibility for firing the timeout. Should it be `getTxnResult` or `coordinateAsync`?

I am thinking the highest level is where we should do it because that is where we are blocking and Accord might fail to fire the timeout?

Reply:
An now I remember why this was tricky, the issue is how I did CAS where it maps the future and never calls `getTxnResult`. I need to fix that so it goes through the `getTxnResult` code so their share this error handling and metrics.

Reply:
I just didn't know what `j.u.c` meant, although now I realize it's `java.util.concurrent`